docs/rfds/session-config-options.mdx

```diff
@@ -0,0 +1,240 @@
+          }
+        ]
+      },
+      {
+        "id": "models",
+        "name": "Model",
+        "currentId": "ask",
+        "options": [
+          {
+            "id": "model-1",
+            "name": "Model 1",
+            "description": "The fastest model"
+          },
+          {
+            "id": "model-2",
+            "name": "Model 2",
+            "description": "The most powerful model"
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+When we introduce this, we could also allow for grouped options, in case there are logical sub-headers and groupings for the options in an individual selector. It
+
+```json
+{
+  "id": "models",
+  "name": "Model",
+  "currentId": "ask",
+  "options": [
+    {
+      "group": "Provider A",
+      "options": [
+        {
+          "id": "model-1",
+          "name": "Model 1",
+          "description": "The fastest model"
+        }
+      ]
+    },
+    {
+      "group": "Provider B",
+      "options": [
+        {
+          "id": "model-2",
+          "name": "Model 2",
+          "description": "The most powerful model"
+        }
+      ]
+    }
+  ]
+}
+```
+
+We use a list of objects for all of these, to ensure consistent ordering of both the config options and the possible values across languages that may or may not preserve ordering.
+
+For grouping options, it needs to be explored whether or not grouped and ungrouped options can be interspersed, or if we need to restrict to one mode or the other (likely the latter).
+
+For updating the value from the client and agent, it would follow the same pattern as [session modes](https://agentclientprotocol.com/protocol/session-modes#from-the-client) but have an additional key.
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 2,
+  "method": "session/set_config_option",
+  "params": {
+    "sessionId": "sess_abc123def456",
+    "configId": "mode",
+    "optionId": "code"
+  }
+}
+```
+
+And the response to this request would return the full list of config options with current values.
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 2,
+  "result": {
+    "config_options": [
+      {
+        "id": "mode",
+        "name": "Session Mode",
+        "currentId": "ask",
+        "options": [..]
+      },
+      {
+        "id": "models",
+        "name": "Model",
+        "currentId": "ask",
+        "options": [..]
+      }
+    ]
+  }
+}
+```
+
+The notification would return the full list of config options with current values as well.
+
+```json
+{
+  "jsonrpc": "2.0",
+  "method": "session/update",
+  "params": {
+    "sessionId": "sess_abc123def456",
+    "update": {
+      "sessionUpdate": "config_option_update",
+      "config_options": [
+        {
+          "id": "mode",
+          "name": "Session Mode",
+          "currentId": "ask",
+          "options": [..]
+        },
+        {
+          "id": "models",
+          "name": "Model",
+          "currentId": "ask",
+          "options": [..]
+        }
+      ]
+    }
+  }
+}
+```
```

Comment:
I think we could simplify the Client-side implementation by only including the config options state in this notification, and not in the `session/set_config_option`. The rationale being that the Client is gonna _have_ to handle the notification anyway for changes originating from the Agent, so we could have just have one source of these. 

Let me know if you disagree, though.

Reply:
So you're suggesting, if I set an value, I just get a null back and rely on the agent sending an update if other values have changed as a result?